shell/platform/darwin/macos/framework/Source/FlutterPlatformNodeDelegateMac.mm

```diff
@@ -0,0 +1,105 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterPlatformNodeDelegateMac.h"
+
+#import "flutter/shell/platform/darwin/macos/framework/Headers/FlutterAppDelegate.h"
+#import "flutter/shell/platform/darwin/macos/framework/Headers/FlutterViewController.h"
+#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterEngine_Internal.h"
+
+#include "flutter/shell/platform/common/accessibility_bridge.h"
+#include "flutter/third_party/accessibility/ax/ax_action_data.h"
+#include "flutter/third_party/accessibility/ax/ax_node_position.h"
+#include "flutter/third_party/accessibility/ax/platform/ax_platform_node.h"
+#include "flutter/third_party/accessibility/ax/platform/ax_platform_node_base.h"
+#include "flutter/third_party/accessibility/base/string_utils.h"
+#include "flutter/third_party/accessibility/gfx/geometry/rect_conversions.h"
+
+namespace flutter {  // namespace
+
+FlutterPlatformNodeDelegateMac::FlutterPlatformNodeDelegateMac(FlutterEngine* engine)
+    : engine_(engine) {
+  ax_platform_node_ = ui::AXPlatformNode::Create(this);
+  NSCAssert(ax_platform_node_, @"Failed to create platform node.");
+}
+
+FlutterPlatformNodeDelegateMac::~FlutterPlatformNodeDelegateMac() {
+  ax_platform_node_->Destroy();
+}
+
+gfx::NativeViewAccessible FlutterPlatformNodeDelegateMac::GetNativeViewAccessible() {
+  NSCAssert(ax_platform_node_, @"Platform node does not exist.");
+  return ax_platform_node_->GetNativeViewAccessible();
+}
+
+gfx::NativeViewAccessible FlutterPlatformNodeDelegateMac::GetParent() {
+  gfx::NativeViewAccessible parent = FlutterPlatformNodeDelegate::GetParent();
+  if (!parent) {
+    NSCAssert(engine_, @"Flutter engine should not be deallocated");
+    return engine_.viewController.view;
+  }
+  return parent;
+}
+
+gfx::Rect FlutterPlatformNodeDelegateMac::GetBoundsRect(
+    const ui::AXCoordinateSystem coordinate_system,
+    const ui::AXClippingBehavior clipping_behavior,
+    ui::AXOffscreenResult* offscreen_result) const {
+  gfx::Rect bounds = FlutterPlatformNodeDelegate::GetBoundsRect(
+      coordinate_system, clipping_behavior, offscreen_result);
+  // Applies window transform.
+  NSRect local_bounds;
+  local_bounds.origin.x = bounds.x();
+  local_bounds.origin.y = bounds.y();
+  local_bounds.size.width = bounds.width();
+  local_bounds.size.height = bounds.height();
+  // local_bounds is flipped against y axis.
```

Comment:
Since we have a mix of platform experience on the team it would be good to provide the tiny bit of additional context that macOS Y co-ords start at bottom-left and increase towards top-right vs Flutter's top-left increasing to bottom-right.